src/bin/wit-bindgen.rs

```diff
@@ -283,7 +283,21 @@ fn gen_world(
         exports,
         default: world.default,
     };
-    generator.generate(&world.name, &interfaces, files);
+    let name = match &world.name {
+        Some(name) => name,
+        None => {
+            if let Some(default) = &interfaces.default {
+                &default.name
+            } else if interfaces.exports.len() > 0 {
+                &interfaces.exports.first().unwrap().1.name
+            } else if interfaces.imports.len() > 0 {
+                &interfaces.imports.first().unwrap().1.name
+            } else {
+                return Err(anyhow!("no interfaces provided to generate"));
+            }
```

Comment:
Eventually my intention was to make the `--name` flag optional and inferred from the "world" where eventually a world will be specified all in one file or the default world would have a name. For example I'd expect the invocation to look like `wit-bindgen guest c foo.wit` would infer the name "foo" for the generated bindings and the `--import`/`--export`/`--default` flags would all go away.

That's all for the future, though, and it seems reasonable to remove the required `--name` for now. That being said I'm a little uncomfortable just picking the first export or import, so could this require either `--name` or `--default`, otherwise returning an error saying that one of those must be specified?

Reply:
Yeah, understood this will all change, but nice to keep the overall CLI expectations. Sure, added!